pkg/controlplane/controller/leaderelection/election.go

```diff
@@ -39,72 +39,27 @@ func pickBestLeaderOldestEmulationVersion(candidates []*v1alpha1.LeaseCandidate)
 	return electee
 }
 
-// topologicalSortWithOneRoot has a caveat that there may only be one root (indegree=0) node in a valid ordering.
-func topologicalSortWithOneRoot(graph map[v1.CoordinatedLeaseStrategy][]v1.CoordinatedLeaseStrategy) []v1.CoordinatedLeaseStrategy {
-	inDegree := make(map[v1.CoordinatedLeaseStrategy]int)
-	for node := range graph {
-		inDegree[node] = 0
-	}
-	for _, neighbors := range graph {
-		for _, neighbor := range neighbors {
-			inDegree[neighbor]++
-		}
-	}
-
-	var queue []v1.CoordinatedLeaseStrategy
-	for vertex, degree := range inDegree {
-		if degree == 0 {
-			queue = append(queue, vertex)
-		}
-	}
-
-	// If multiple nodes have indegree of 0, multiple strategies are non-superceding and is a conflict.
-	if len(queue) > 1 {
-		return nil
-	}
-
-	var sorted []v1.CoordinatedLeaseStrategy
-	for len(queue) > 0 {
-		vertex := queue[0]
-		queue = queue[1:]
-		sorted = append(sorted, vertex)
-
-		for _, neighbor := range graph[vertex] {
-			inDegree[neighbor]--
-			if inDegree[neighbor] == 0 {
-				queue = append(queue, neighbor)
-			}
-		}
-	}
-
-	if len(sorted) != len(graph) {
-		return nil // Cycle detected
-	}
-
-	return sorted
-}
-
-func pickBestStrategy(candidates []*v1alpha1.LeaseCandidate) (v1.CoordinatedLeaseStrategy, error) {
-	graph := make(map[v1.CoordinatedLeaseStrategy][]v1.CoordinatedLeaseStrategy)
+func pickBestStrategy(candidates []*v1alpha2.LeaseCandidate) (v1.CoordinatedLeaseStrategy, error) {
 	nilStrategy := v1.CoordinatedLeaseStrategy("")
-	for _, c := range candidates {
-		for i := range len(c.Spec.PreferredStrategies) - 1 {
-			graph[c.Spec.PreferredStrategies[i]] = append(graph[c.Spec.PreferredStrategies[i]], c.Spec.PreferredStrategies[i+1])
-		}
-		if _, ok := graph[c.Spec.PreferredStrategies[len(c.Spec.PreferredStrategies)-1]]; !ok {
-			graph[c.Spec.PreferredStrategies[len(c.Spec.PreferredStrategies)-1]] = []v1.CoordinatedLeaseStrategy{}
-		}
+	if len(candidates) == 0 {
+		return nilStrategy, fmt.Errorf("no candidates")
 	}
-
-	sorted := topologicalSortWithOneRoot(graph)
-	if sorted == nil {
-		return nilStrategy, fmt.Errorf("invalid strategy")
+	candidateName := candidates[0].Name
+	strategy := candidates[0].Spec.Strategy
+	highestBV := candidates[0].Spec.BinaryVersion
+
+	for _, c := range candidates[1:] {
+		if c.Spec.BinaryVersion > highestBV {
```

Comment:
is this the right order? `v1.1.2 > v1.11.10` which is wrong.

Reply:
You're right that this should be a semver comparison and I've updated  though I'm having trouble finding a counterexample. Lexicographically numbers are greater than dots (.) so `v1.1.2 < v1.11.10`.